cub/test/test_device_spmv.cu

```diff
@@ -121,7 +123,8 @@ struct csr_matrix
 
   void finalize()
   {
-    thrust::exclusive_scan(m_row_offsets.cbegin(), m_row_offsets.cend(), m_row_offsets.begin());
+    using exec_policy_t = cub::detail::conditional_t<HostStorage, decltype(thrust::host), c2h::device_policy_t>;
+    thrust::exclusive_scan(exec_policy_t{}, m_row_offsets.cbegin(), m_row_offsets.cend(), m_row_offsets.begin());
```

Comment:
I'm not sure how this is compiling 😕

`c2h::device_policy_t` is an alias for

```
thrust::detail::execute_with_allocator<
  c2h::checked_cuda_allocator<char>, 
  thrust::cuda_cub::execute_on_stream_base
>
```

and `execute_with_allocator` [doesn't have a default ctor](https://github.com/NVIDIA/cccl/blob/main/thrust/thrust/detail/execute_with_allocator_fwd.h#L46-L54).

Maybe I'm missing something, but it seems like this should be:

```suggestion
    _CCCL_IF_CONSTEXPR(HostStorage)
    {
      thrust::exclusive_scan(thrust::host, m_row_offsets.cbegin(), m_row_offsets.cend(), m_row_offsets.begin());
    }
    else
    {
      thrust::exclusive_scan(c2h::device_policy, m_row_offsets.cbegin(), m_row_offsets.cend(), m_row_offsets.begin());
    }   
```

Reply:
Maybe `HostStorage` is always true?

Reply:
Nope, `device_csr_matrix` is used in the test.

Huh.

Reply:
It seems `csr_matrix::finalize()` is never called when `HostStorage` is false. I am not sure this is legal C++, since `finalize` does not have any template parameters, so I think it must be instantiated (and checked) when `csr_matrix` is instantiated.